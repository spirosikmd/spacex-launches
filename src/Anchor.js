import React from 'react';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  anchor: {
    color: theme.palette.primary.light,
    '&:hover': {
      color: theme.palette.primary.main,
    },
    '&:visited': {
      color: theme.palette.primary.dark,
    },
  },
});

export const Anchor = ({ classes, children, ...props }) => {
  return (
    <a className={classes.anchor} {...props}>
      {children}
    </a>
  );
};

export default withStyles(styles)(React.memo(Anchor));
