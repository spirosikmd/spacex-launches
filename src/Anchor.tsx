import React, { ReactNode } from 'react';
import {
  withStyles,
  Theme,
  WithStyles,
  createStyles,
} from '@material-ui/core/styles';

const styles = (theme: Theme) =>
  createStyles({
    anchor: {
      color: theme.palette.primary.light,
      '&:hover': {
        color: theme.palette.primary.main,
      },
      '&:visited': {
        color: theme.palette.primary.dark,
      },
    },
  });

export interface AnchorProps extends WithStyles<typeof styles> {
  children: ReactNode;
  target: string;
  href: string;
}

const Anchor = ({ classes, children, ...props }: AnchorProps) => {
  return (
    <a className={classes.anchor} {...props}>
      {children}
    </a>
  );
};

export default withStyles(styles)(React.memo(Anchor));
