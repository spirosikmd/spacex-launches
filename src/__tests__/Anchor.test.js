import Anchor from '../Anchor';

describe('Anchor', () => {
  let props;

  beforeEach(() => {
    props = {
      classes: {
        anchor: 'anchor',
      },
      href: 'href',
      target: 'target',
      children: 'text',
    };
  });

  it('renders', () => {
    expect(mountComponent(Anchor, props)).toMatchSnapshot();
  });
});
