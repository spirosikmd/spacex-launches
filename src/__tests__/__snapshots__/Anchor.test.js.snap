// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor renders 1`] = `
<WithStyles(Anchor)
  classes={
    Object {
      "anchor": "anchor",
    }
  }
  href="href"
  target="target"
>
  <a
    className="Anchor-anchor-1 anchor"
    href="href"
    target="target"
  >
    text
  </a>
</WithStyles(Anchor)>
`;
